src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/TensorPrimitives.cs

```diff
@@ -1306,12 +1238,25 @@ public static void Tanh(ReadOnlySpan<float> x, Span<float> destination)
                 ThrowHelper.ThrowArgument_DestinationTooShort();
             }
 
+            ValidateInputOutputSpanNonOverlapping(x, destination);
+
             for (int i = 0; i < x.Length; i++)
             {
                 destination[i] = MathF.Tanh(x[i]);
             }
         }
 
+        /// <summary>Throws an exception if the <paramref name="input"/> and <paramref name="output"/> spans overlap and don't begin at the same memory location.</summary>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static void ValidateInputOutputSpanNonOverlapping(ReadOnlySpan<float> input, Span<float> output)
+        {
+            if (!Unsafe.AreSame(ref MemoryMarshal.GetReference(input), ref MemoryMarshal.GetReference(output)) &&
```

Comment:
so there are cases where it's ok to have input and output to be the same reference? Is it true for all APIs?

Reply:
btw, I'll take a look what we can improve in codegen for `MemoryExtensions.Overlaps`

Reply:
> so there are cases where it's ok to have input and output to be the same reference? Is it true for all APIs?

Yes, all of the APIs support in-place now.